traffic_ops/experimental/go-api/tenant/EndPointSeeder.go

```diff
@@ -0,0 +1,202 @@
+package tenant
+
+/*
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+ 
+import (
+    "database/sql"
+    "encoding/json"
+    "net/http"
+    "fmt"
+    "time"
+
+    "github.com/gorilla/mux"
+  _ "github.com/lib/pq"
+)
+
+
+//TODO(nirs) polymorphisem
+
+/////BASE - COMMON TO ALL SEEDERS (needs to move to another package)
+/// Defintions
+type EndpointSeeder struct{
+    myDb        *sql.DB
+}
+
+/// Interface
+func NewEndpointSeeder(aDb *sql.DB) *EndpointSeeder {
+    endPointSeeder := new(EndpointSeeder)
+    endPointSeeder.myDb = aDb
+    return endPointSeeder
+}
+
+func (aEndpointSeeder *EndpointSeeder) Seed (aRouter *mux.Router, aApiPrefix string) {
+
+    aRouter.HandleFunc(aApiPrefix+"/tenants", aEndpointSeeder.getListEndpoint).Methods("GET")
+    aRouter.HandleFunc(aApiPrefix+"/tenants/{id}", aEndpointSeeder.getEndpoint).Methods("GET")
+    aRouter.HandleFunc(aApiPrefix+"/tenants", aEndpointSeeder.createEndpoint).Methods("POST")
+    aRouter.HandleFunc(aApiPrefix+"/tenants/{id}", aEndpointSeeder.updateEndpoint).Methods("PUT")
+    aRouter.HandleFunc(aApiPrefix+"/tenants/{id}", aEndpointSeeder.deleteEndpoint).Methods("DELETE")
+}
+
+///Common utilities
+func (aEndpointSeeder *EndpointSeeder) checkErr(aError error) {
+    //TODO log
+    if aError != nil {
+        panic(aError)
+    }
+}
+
+
+/////Class specific
+
+type Tenant struct {
+    Id        int       `json:"id"`
+    Name      string    `json:"name"`
+    Active    bool      `json:"active"`
+    ParentId  int       `json:"parentId"`
+}
+
+func (aEndpointSeeder *EndpointSeeder) getEndpoint(aResponseWriter http.ResponseWriter, aRequest *http.Request) {
+    params := mux.Vars(aRequest)
+
+    var id int
+    var name string
+    var active bool
+    var parent_id sql.NullInt64
+    var updated time.Time;
+    
+    err := aEndpointSeeder.myDb.QueryRow("SELECT * FROM tenant WHERE id="+params["id"]).Scan(&id, &name, &active, &parent_id, &updated)
+    if err == sql.ErrNoRows{//TODO make it work
+        aResponseWriter.WriteHeader(http.StatusNotFound)
+        fmt.Fprint(aResponseWriter, params["id"]+" not found")
```

Comment:
`params["id"]` Is used a lot in this function. Consider putting it in a variable?